packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,257 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+
+import 'basic_types.dart';
+import 'text_style.dart';
+import 'text_painter.dart';
+
+/// An immutable span of inline content which form a paragraph.
+///
+/// The subclass [TextSpan] specifies text and may contain child [InlineSpan]s.
+///
+/// The subclass [WidgetSpan] specifies embedded inline widgets. Specify a
+/// widget by wrapping the widget with a [WidgetSpan]. The widget will be
+/// laid out inline within the paragraph.
```

Comment:
we should also mention PlaceholderSpan here. For example, say how if you're creating a new library that uses TextPainter, you may want to use a different kind of Placeholder, and that in the widgets library, we use WidgetSpan (then explain WidgetSpan as you do above).